api/bin/chainflip-broker-api/src/main.rs

```diff
@@ -194,6 +219,62 @@ impl RpcServer for RpcServerImpl {
 			)
 			.await?)
 	}
+
+	async fn mark_transaction_as_tainted(&self, tx_id: TransactionInId) -> RpcResult<()> {
+		self.api
+			.deposit_monitor_api()
+			.mark_transaction_as_tainted(tx_id)
+			.await
+			.map_err(BrokerApiError::Other)?;
+		Ok(())
+	}
+
+	async fn get_open_deposit_channels(
+		&self,
+		query: GetOpenDepositChannelsQuery,
+	) -> RpcResult<ChainAccounts> {
+		let account_id = match query {
+			GetOpenDepositChannelsQuery::All => None,
+			GetOpenDepositChannelsQuery::Mine => Some(self.api.state_chain_client.account_id()),
+		};
+
+		self.api
+			.state_chain_client
+			.base_rpc_client
+			.raw_rpc_client
+			.cf_get_open_deposit_channels(account_id, None)
+			.await
+			.map_err(BrokerApiError::ClientError)
+	}
+
+	async fn subscribe_tainted_transaction_events(
+		&self,
+		pending_sink: PendingSubscriptionSink,
+	) -> SubscriptionResult {
+		let subscription_stream = self
```

Comment:
same as subscription in the lp-api, this should probably used the finalised stream from the sc_client since we then have a guarantee that we can't skip blocks

Reply:
Done.